lib/Monad_Lib.thy

```diff
@@ -0,0 +1,64 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+theory Monad_Lib
```

Comment:
what's the guideline for what goes here?

Reply:
Good point, that should go into a comment. This theory should be the minimal set of constants and lemmas to support the monad definitions.